### PR TITLE
[MI-3396] Fixed the logging messages everywhere in the plugin along with the unit tests

### PR DESCRIPTION
**Pull request by manojmalik20**
#### Summary
- Fixed the logging messages everywhere
- Modified the unit tests

### CODE REVIEW

File: server/handlers/getters.go [hunk 1]
```diff
@@ -34,14 +34,14 @@ func (ah *ActivityHandler) getMessageFromChat(chat *msteams.Chat, messageID stri
...
-		ah.plugin.GetAPI().LogError("unable to get bot client", "error", err)
+		ah.plugin.GetAPI().LogError("unable to get client for user", "userID", userID, "error", err)
 		return nil, err
 	}
 
 	var msg *msteams.Message
 	msg, err = client.GetReply(teamID, channelID, messageID, replyID)
 	if err != nil {
-		ah.plugin.GetAPI().LogError("Unable to get original post", "error", err)
+		ah.plugin.GetAPI().LogError("Unable to get reply from channel", "error", err)
```
Comment (ayusht2810) [inline server/handlers/getters.go:44]: should we log some additional info on the post?

File: server/handlers/getters_test.go [hunk 2]
```diff
@@ -587,15 +592,18 @@ func TestGetMessageAndChatFromActivityIds(t *testing.T) {
...
 				p.On("GetClientForUser", testutils.GetUserID()).Return(client, nil)
 				p.On("GetAPI").Return(mockAPI)
 			},
-			setupClient: func() {
+			setupClient: func(client *mocksClient.Client) {
 				client.On("GetReply", testutils.GetTeamsUserID(), testutils.GetChannelID(), testutils.GetMessageID(), testutils.GetReplyID()).Return(nil, errors.New("Error while getting reply from channel"))
 			},
-			setupStore: func() {
+			setupAPI: func(api *plugintest.API) {
+				api.On("LogError", "Unable to get reply from channel", "error", errors.New("Error while getting reply from channel"))
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/handlers/getters_test.go:604]: fix everywhere, if changing

File: server/api.go [hunk 3]
```diff
@@ -55,7 +55,7 @@ func (a *API) getAvatar(w http.ResponseWriter, r *http.Request) {
 		var err error
 		photo, err = a.p.msteamsAppClient.GetUserAvatar(userID)
 		if err != nil {
-			a.p.API.LogError("Unable to read avatar", "error", err.Error())
+			a.p.API.LogError("Unable to get user avatar", "userID", userID, "error", err.Error())
```
Comment (Kshitij-Katiyar) [inline server/api.go:58]: try to segregate whether this a mmuser or teamsuser

File: server/api.go [hunk 4]
```diff
@@ -125,7 +125,7 @@ func (a *API) refreshSubscriptionIfNeeded(activity msteams.Activity) {
 			a.p.API.LogError("Unable to refresh the subscription", "error", err.Error())
 		} else {
 			if err2 := a.p.store.UpdateSubscriptionExpiresOn(activity.SubscriptionID, *expiresOn); err2 != nil {
```
Comment (Kshitij-Katiyar) [inline server/api.go:127]:
```suggestion
			if err := a.p.store.UpdateSubscriptionExpiresOn(activity.SubscriptionID, *expiresOn); err != nil {
```

File: server/api.go [hunk 5]
```diff
@@ -172,7 +172,7 @@ func (a *API) autocompleteTeams(w http.ResponseWriter, r *http.Request) {
 
 	client, err := a.p.GetClientForUser(userID)
 	if err != nil {
-		a.p.API.LogError("Unable to get the client for user", "Error", err.Error())
+		a.p.API.LogError("Unable to get the client for user", "UserID", userID, "Error", err.Error())
```
Comment (Kshitij-Katiyar) [inline server/api.go:175]: same if possible

File: server/api.go [hunk 6]
```diff
@@ -334,8 +334,8 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 
 	mmUserID := stateArr[1]
 	if err := a.store.VerifyOAuth2State(state); err != nil {
-		a.p.API.LogError("Unable to complete OAuth.", "UserID", mmUserID, "Error", err.Error())
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		a.p.API.LogError("Unable to verify OAuth state", "UserID", mmUserID, "Error", err.Error())
+		http.Error(w, "Unable to complete OAuth2.", http.StatusInternalServerError)
```
Comment (Kshitij-Katiyar) [inline server/api.go:338]:
```suggestion
		http.Error(w, "Unable to complete authorisation.", http.StatusInternalServerError)
```

File: server/message_hooks_test.go [hunk 7]
```diff
@@ -933,7 +933,7 @@ func TestUnsetReaction(t *testing.T) {
 		{
 			Name: "UnsetReaction: Unable to unset the reaction",
 			SetupAPI: func(api *plugintest.API) {
-				api.On("LogWarn", "Error creating post", "error", mock.Anything)
+				api.On("LogWarn", "Error in removing the reaction", "emojiName", "mockName", "error", ": , unable to unset the reaction")
```
[the remark this answers is not in the document]
Comment (Kshitij-Katiyar) [reply server/message_hooks_test.go:936]: please check this everywhere

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by ayusht2810 — APPROVED** [review]

**Pull request merged**